mlflow/tracing/fluent.py

```diff
@@ -177,15 +177,15 @@ def start_span(
             trace_manager = InMemoryTraceManager.get_instance()
             request_id = trace_manager.get_or_create_request_id(span.context.trace_id)
             # Setting end_on_exit = False to suppress the default span
-            # export and instead invoke MlflowSpanWrapper.end()
+            # export and instead invoke Span.end()
             with trace_api.use_span(span, end_on_exit=False):
-                mlflow_span = MlflowSpanWrapper(span, request_id=request_id, span_type=span_type)
+                mlflow_span = LiveSpan(span, request_id=request_id, span_type=span_type)
                 mlflow_span.set_attributes(attributes or {})
                 trace_manager.add_or_update_span(mlflow_span)
                 yield mlflow_span
         else:
             # Span creation should not raise an exception
```

Comment:
Why shouldn't it raise an exception? Feel free to file a follow-up ticket to address this. I think it should throw.

Reply:
Imo instrumentation logic should not bother the main application logic - it would be annoying in especially served model - for many case it is acceptable not to see the trace but returning prediction error would not. WDYT?
https://docs.google.com/document/d/15iOQicMz8hP6-VsVLs_l26TrDMNoVd_zb-tWRcTUtgo/edit#heading=h.yx9scpaqor2w

Reply:
You’re right. Please disregard.